cmd/state-exec/main.go

```diff
@@ -65,10 +65,22 @@ func run() error {
 	}
 	logr.Debug("message data - pid: %s, exec: %s", hb.ProcessID, hb.ExecPath)
 
+	a, err := newAttempt()
+	if err != nil {
+		return fmt.Errorf("cannot create new attempt: %w", err)
+	}
+	logr.Debug("message data - exec: %s", a.ExecPath)
+
 	meta, err := newExecutorMeta(hb.ExecPath)
+	// Send attempt event regardless of whether we can get the meta data.
+	logr.Debug("communications - sock: %s", meta.SockPath)
+	if msgErr := sendMsgToService(meta.SockPath, a); msgErr != nil {
```

Comment:
This is a bit confusing. If acquiring the meta data fails due to not finding the meta file, then the sock path value will not be set. I'm not sure that there is a great way to report an attempt if acquiring the meta data fails aside from communicating directly with whatever data store holds the usage info (i.e. bypassing the service).

There is a small chance that the data is partially corrupt, and does contain the sockpath, but it's a very narrow case.

Reply:
This crossed my mind as well but at the time I decided to not address it. I'll leave it up to @Naatan if he wants us to come up with some way to deal with this situation.

Reply:
Not sure if I'm fully in the loop. But sending it without the metadata is pointless. Without the metadata we can't do anything with the event.

Reply:
This doesn't seem to be addressed.  I think that this approach effectively means that the ExecutorMeta error is being ignored. I suggest printing the communication debug and sending the msg to svc within a conditional block of `if meta.SockPath != "" {`.